### PR TITLE
Zadanie 2

### PR DESCRIPTION
**Pull request by sz3kz**
Rozwiązanie Zadania 2

Rozwiązanie mieści się w `ML_zadanie.ipynb` + png wizualizacji Optuny

NOTE: Niby notebook jest długi, ale większość notebooka to error-output z fine-tuningu optimą modelu regresji logistycznej.

### CODE REVIEW

**Review by milenakulik — COMMENTED** [review]
Cześć, z kabiną fajny pomysł z pierwszą literą, ale od siebie polecam spróbować zwykłej zmiennej binarnej Has_Cabin (1/0). Sam brak kabiny mocno obniżał szanse na przeżycie. Wydaje mi się, że w Optunie możesz mieć data leakage. Dajesz jej chyba cały dataframe_data, a nie tylko dane treningowe.

**Comment by sz3kz** [comment]
Odnośnie dataleak'u to rzeczywiście w wielu miejscach pomyliłem `data_training` i `target_training` z `dataframe_data` i `dataframe_target`.

**Comment by Piotr-Wojdas** [comment]
Ogólnie pracka fajna, duży plus za użycie optuny, tutaj faktycznie była trochę zbędna i można było hiperparametry ustawić "na oko" albo po prostu ręcznie sprawdzić kilka i wybrać ten z najlepszym wynikiem. Z rzeczy do których chcę się przyczepić to learning rate, który ustawiłeś do sprawdzenia aż do wartości 3, raczej lr jest w zakresie od (0,1) i wątpię że optuna dałaby ci lepszy wyniki z lr >1 więc niepotrzebnie wydłużyłeś jej pracę ale to taki szczegół na przyszłość. Reszta raczej git, zadanie zaliczone, sory że tak późno.
